pyserini/search/lucene/ltr/__main__.py

```diff
@@ -206,27 +207,39 @@ def output(file, dev_data, format):
                 score_tie_counter += 1
                 score_tie_query.add(qid)
             prev_score = t.score
-            rank += 1
-            if (format == 'tsv'):
-                output_file.write(f"{qid}\t{t.pid}\t{rank}\n")
+            if (maxp):
+                score = float(score)
+                docid = t.pid.split('#')[0]
+                if (qid not in results or docid not in results[qid] or score > results[qid][docid]):
+                    results[qid][docid] = t.score
             else:
-                output_file.write(f"{qid}\tQ0\t{t.pid}\t{rank}\t{t.score}\tltr\n")
-
+                results[qid][t.pid] = t.score
+
+        for qid, docid_score in tqdm(scores.items()):
+            rank = 1
+            docid_score = sorted(docid_score.items(),key=lambda kv: kv[1], reverse=True)
+            for docid, score in docid_score:
+                if (format=='trec'):
+                    output_file.write(f"{qid}\tQ0\t{docid}\t{rank}\t{score}\tltr\n")
+                else:
+                    output_file.write(f"{qid}\t{docid}\t{rank}\n")
+                rank += 1
     score_tie = f'score_tie occurs {score_tie_counter} times in {len(score_tie_query)} queries'
     print(score_tie)
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description='Learning to rank reranking')
     parser.add_argument('--input', required=True)
-    parser.add_argument('--reranking-top', type=int, default=1000)
+    parser.add_argument('--reranking-top', type=int, default=10000)
```

Comment:
Can you use the same args as the main searcher?
https://github.com/castorini/pyserini/blob/master/pyserini/search/lucene/__main__.py#L112